pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SignatureDeclareThrowsExceptionRule.java

```diff
@@ -77,7 +77,7 @@ public Object visit(ASTClassOrInterfaceDeclaration node, Object data) {
             }
         }
         if (node.jjtGetNumChildren() != 0 && node.jjtGetChild(0) instanceof ASTExtendsList) {
```

Comment:
This block and the one checking the interfaces could be simplified using the methods `getSuperInterfacesTypeNodes` and  `getSuperClassTypeNode` of `ASTClassOrInterfaceDeclaration`